man/homectl.xml

```diff
@@ -356,6 +356,19 @@
         generally do not required that, and work out of the box.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--fido2-credential-algorithm=</option><replaceable>STRING</replaceable></term>
+        <listitem><para>Specify COSE algorithm used in credential generation. The default value is
+        <literal>ES256</literal>. Supported values are <literal>ES256</literal>, <literal>RS256</literal>
+        and <literal>EDDSA</literal>.</para>
+
+        <para><literal>ES256</literal> denotes ECDSA over NIST P-256 with SHA-256. <literal>RS256</literal>
+        denotes 2048-bit RSA with PKCS#1.5 padding and SHA-256. <literal>EDDSA</literal> denotes
+        EDDSA over Curve25519 with SHA-512.</para>
+
+        <para>Note that your authenticator may not support some algorithms.</para></listitem>
```

Comment:
is there a way to enumerate this?

Reply:
Yes, we can use `fido_cbor_info_algorithm_cose` to enumerate supported algorithm on device. I'll going to implement it.

Reply:
that'd certainly be welcome though not necessary, i guess.

question is were to add this. maybe just query it in verify_features() and show it as debug info like the other bits of info we show there right now?

Reply:
Yes, I think it's not necessary as if the authenticator does't support the specified algorithm, `fido_dev_make_cred` will fail with `FIDO_ERR_UNSUPPORTED_ALGORITHM`.

Reply:
After some search I found the function will only work for authenticator that support CTAP 2.1. There is no way to emuerate supported algorithms on authenticator before CTAP 2.1. `authenticatorGetInfo` method won't return `algorithms` field before that version.

Reference:
CTAP 2.0: https://fidoalliance.org/specs/fido-v2.0-rd-20170927/fido-client-to-authenticator-protocol-v2.0-rd-20170927.html#authenticatorgetinfo-0x04
CTAP 2.1: https://fidoalliance.org/specs/fido-v2.1-rd-20210309/fido-client-to-authenticator-protocol-v2.1-rd-20210309.html#authenticatorGetInfo

Reply:
ok, it would be the cherry on top only anyway